### PR TITLE
edward: Lion beta1=0.95 sweep (beta1 never tested above 0.9)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Lion's momentum parameter `beta1` controls how strongly the optimizer uses accumulated gradient direction vs the current gradient. The current SOTA uses `beta1=0.9` (default). We tested `beta1=0.8` in an earlier round and it regressed. The natural next step is to sweep **upward** from 0.9 toward 0.95 — more momentum means slower direction change per step, which can act as an implicit smoother and reduce oscillation in noisy gradient regimes like CFD point clouds.

The full beta1 sweep so far: 0.8 (regression), **0.9 (SOTA)**. The 0.95 point is unexplored and represents the last meaningful test in this axis before closing it.

Reference: Lion paper (Chen et al., 2023) uses beta1=0.9 as default for image classification but notes that values close to 1.0 reduce gradient noise. CFD surrogates have high spatial gradient variance — slightly higher beta1 may stabilize the sign-based update direction.

## Instructions

Run the current SOTA stack with `--lion-beta1 0.95`. Everything else is identical to the SOTA reproduce command.

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent edward \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --lion-beta1 0.95 --lion-beta2 0.99 \
  --no-compile-model \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --ema-decay 0.999 --lr-warmup-epochs 1 \
  --wandb-group tay-lion-beta1-sweep
```

Key change: `--lion-beta1 0.95` (was 0.9 in SOTA). All other flags match current SOTA exactly.

Report: best val_abupt and the epoch it occurred at, total training time, and all test_primary/* metrics from the best-val checkpoint.

## Baseline (PR #232 askeladd, W&B run `r8s2dtnq`)

| Metric | Current SOTA (PR #232, beta1=0.9) | AB-UPT |
|--------|----------------------------------:|-------:|
| `val_abupt` (best) | **9.0650%** (ep11) | — |
| `test_abupt` | **10.190%** | — |
| `surface_pressure` | 5.461% | 3.82% |
| `wall_shear` | 9.910% | 7.29% |
| `volume_pressure` | 12.656% | 6.08% |
| `tau_x` | 8.432% | 5.35% |
| `tau_y` | 11.952% | 3.65% |
| `tau_z` | 12.447% | 3.63% |

Beat target: val_abupt < **9.0650%** and/or test_abupt < **10.190%**.

Prior beta1 results for context:
- beta1=0.8: regressed (higher val_abupt than SOTA)
- beta1=0.9: current SOTA
- beta1=0.95: this experiment

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Your pod has been running ~9min. When you get your first epoch val result, post the W&B run ID here. We're watching beta1=0.95 vs SOTA beta1=0.9 — report both training loss trajectory and first val metric as soon as available.

**Comment by morganmcg1** [comment]
STUDENT edward:

W&B run ID: `dv107rp0` (rank 0) — group `tay-lion-beta1-sweep`
URL: https://wandb.ai/wandb-applied-ai-team/senpai-v1-drivaerml-ddp8/runs/dv107rp0

**Epoch 1 (warmup epoch, LR still ramping)**
- train_loss=0.92725
- val_abupt_axis_rel_l2_pct=**63.7053%**
- val_surface loss=0.49807, surface_p_mae=0.14569, volume_p_mae=99.60746, wall_shear_mae=0.66332
- 1465s wall time, peak ~56.8GB VRAM
- 34 val cases

Comparison: epoch 1 is the warmup epoch (lr_warmup_epochs=1), so val is inflated as expected. Currently in epoch 2 (~14% through). Will continue to update with epoch 2 val once it finishes (~22 min).

**Comment by morganmcg1** [comment]
STUDENT edward:

**Epoch 2 update** (run `dv107rp0`)
- train_loss=0.6978
- val_abupt_axis_rel_l2_pct=**53.3205%** (down from 63.7053% at ep1)
- val_surface_pressure_rel_l2_pct=40.2875%
- val_wall_shear_rel_l2_pct=57.4640%
- val_volume_pressure_rel_l2_pct=34.6648%
- ~1502s wall time per epoch

Currently in epoch 3 (60/2720 batches, ~2%). All slopes negative — model is descending normally. Training continues. Will report again at epoch ~5 and at completion.

For context, SOTA beta1=0.9 hit best val_abupt=9.0650% at epoch 11. With 6h timeout and ~25min per epoch we should reach ~14 epochs.

**Comment by morganmcg1** [comment]
STUDENT edward:

**Epoch 3 update** (run `dv107rp0`)
- val_abupt_axis_rel_l2_pct=**26.1026%** (down from 53.3205% at ep2)
- val_surface_pressure_rel_l2_pct=18.0070%
- val_wall_shear_rel_l2_pct=28.7702%
- val_volume_pressure_rel_l2_pct=16.0867%
- val_loss=0.08163
- best checkpoint updated at ep3
- ~1465s/epoch wall time

**Comparison to SOTA `r8s2dtnq` (beta1=0.9) at same step counts:**
| Epoch | beta1=0.95 (ours) | beta1=0.9 (SOTA) | Gap |
|------:|------------------:|-----------------:|----:|
| 1 (warmup) | 63.71% | 49.52% | +14.2pp |
| 2 | 53.32% | 21.97% | +31.4pp |
| 3 | 26.10% | 15.16% | +11.0pp |

beta1=0.95 is converging but trailing SOTA. Gap narrowed substantially from ep2→ep3 (31.4pp→11.0pp), consistent with the longer EMA effective horizon (~20 steps vs ~10 for beta1=0.9) needing more time to fill the momentum buffer. Still behind, but the slope is encouraging. Currently mid-epoch 4 (~22%). Will report again at ep5.

**Comment by morganmcg1** [comment]
STUDENT edward:

**Epoch 4 update — regression** (run `dv107rp0`)
- train_loss=0.60921 (up from 0.157 at ep3 — 4x increase)
- val_abupt_axis_rel_l2_pct=**79.1957%** (up from 26.10% at ep3 — sharp regression)
- val_surface_pressure_rel_l2_pct=61.7646%
- val_wall_shear_rel_l2_pct=89.8462%
- val_volume_pressure_rel_l2_pct=55.9763%
- best checkpoint stayed at ep3 (26.10%)
- ~1464s/epoch wall time

**Comparison to SOTA `r8s2dtnq` (beta1=0.9) at same epoch counts:**
| Epoch | beta1=0.95 (ours) | train_loss | beta1=0.9 (SOTA) | train_loss |
|------:|------------------:|-----------:|-----------------:|-----------:|
| 1 (warmup) | 63.71% | 0.927 | 49.52% | 0.699 |
| 2 | 53.32% | 0.698 | 21.97% | 0.129 |
| 3 | 26.10% | 0.157 | 15.16% | 0.070 |
| 4 | **79.20%** | **0.609** | 12.53% | 0.047 |

Both train and val regressed sharply at ep4. SOTA showed monotone descent through ep4. The W&B slope summary now reads val_abupt slope=+19.5pp/1k steps and train_loss slope=+0.17/1k steps — both positive (worsening).

Hypothesis: beta1=0.95 momentum buffer (~20-step EMA horizon, vs ~10 for beta1=0.9) is holding stale direction info too long for sign-based updates with the high gradient variance in this regime. Effectively the optimizer kept stepping in a direction that became wrong, and the per-batch grad is too damped (only 5% weight) to course-correct.

Currently in epoch 5 (~25%, 18 min remaining). Will report ep5 to confirm whether this is a transient spike or sustained divergence. If ep5 doesn't recover toward ep3 numbers, beta1=0.95 is clearly worse than SOTA and we have the answer (closing the upward beta1 sweep). Continuing the run to gather full evidence rather than killing early.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #282 because NEGATIVE — Lion beta1=0.95 diverged at epoch 4. Val jumped from 26.1% (ep3) to 79.2% (ep4) and train_loss from 0.157 to 0.609. This is a clear divergence, not noise. The longer EMA horizon (~20-step effective window vs ~10 for beta1=0.9) retains stale gradient direction too long for sign-based updates in this high-variance regime. Lion's optimizer is confirmed to work best with the default beta1=0.9 on this task. W&B run: dv107rp0..
